### PR TITLE
Support VTOL_LAND options

### PR DESCRIPTION
**Pull request by WickedShell**
This allows the GCS to make a `NAV_VTOL_LAND` either do the circle to land or a normal straight in land. It should have always been done this way instead of as a `Q_OPTIONS` bit, but I took the fast path originally.

This also allows us to specify a straight in approach with automatic detransition which we currently aren't supporting. The reason to not use a Q_OPTIONS bit is the same mission may want to use both a spiral landing and a specified approach heading.

### CODE REVIEW

**Comment by tridge** [comment]
see also an alternative approach (pun intended) here: #14938

**Comment by WickedShell** [comment]
Updated, and SITL tested. MAVLink commit is an easy thing to resolve the conflict on, but I'm actually happy with that being a blocker until the change is merged into the sub module.

**Comment by tridge** [comment]
@WickedShell looks good, we just need to get the mavlink changes in first, then ensure the mavlink git hash points to the master HEAD in this PR
thanks!

**Comment by WickedShell** [comment]
Updated the MAVLink submodule now that the MAVLink repo has the message, and fixed the conflicts.

**Comment by tridge** [comment]
@WickedShell failing lots of tests, looks like mavlink not updated correctly?

**Comment by WickedShell** [comment]
> @WickedShell failing lots of tests, looks like mavlink not updated correctly?

Sorry, bad rebasing, I've fixed it.

**Review by tridge — APPROVED** [review]

File: libraries/AP_Soaring/AP_Soaring.cpp
Comment (tridge) [inline libraries/AP_Soaring/AP_Soaring.cpp:169]: these changes are not needed any more as the warnings have been fixed

**Review by tridge — CHANGES_REQUESTED** [review]
ok to merge when soaring changes backed out

**Pull request merged**
